File: languages/js/src/entity/identity_v2/method.ts
import { Method as MethodProto } from "../../bridge/proto/identity_entities_v2";

export enum Method {
  UNRECOGNIZED = -1,
  POLYGON_ID = 0,
  IDEN3 = 1
}

export namespace Method {
  export function toProto(method: Method): MethodProto {
    switch (method) {
      case Method.POLYGON_ID:
        return MethodProto.POLYGON_ID;
      case Method.IDEN3:
        return MethodProto.IDEN3;
      default:
        return MethodProto.UNRECOGNIZED;
    }
  }

  export function fromProto(method: MethodProto | undefined): Method {
    switch (method) {
      case MethodProto.POLYGON_ID:
        return Method.POLYGON_ID;
      case MethodProto.IDEN3:
        return Method.IDEN3;
      default:
        return Method.UNRECOGNIZED;
    }
  }
}

File: languages/python/bloock/entity/identity_v2/method.py
from __future__ import annotations

from enum import Enum
from bloock._bridge.proto.identity_entities_v2_pb2 import Method as MethodProto


class Method(Enum):
    IDEN3 = 0
    POLYGON_ID = 1

    def __int__(self):
        return self.value

    @staticmethod
    def to_proto(method: Method) -> MethodProto.ValueType:
        if method == Method.IDEN3:
            return MethodProto.IDEN3
        elif method == Method.POLYGON_ID:
            return MethodProto.POLYGON_ID
